lesson11/index.js

```diff
@@ -24,31 +20,36 @@ const removeLoadingImage = () => {
 const getListData = () => {
   return new Promise((resolve)=> {
     try {
-      setTimeout(() => {
-        resolve(list);
-      }, 3000);
+      resolve(fetch(url));
     } catch (error) {
       console.error(error);
     } finally {
-      console.log('finally句内です');
+      console.log('finally');
     }
   });
 }
 
-const createList = async ()=> {
-  createLoadingImage();
-  const listData = await getListData();
-  removeLoadingImage();
-  listData.forEach( listDataItem => {
+const createList = async ({data})=> {
+  const fetchedData = data;
+  fetchedData.forEach( element => {
     const li = document.createElement('li');
     const a = document.createElement('a');
-    a.textContent = listDataItem.text;
-    a.href = listDataItem.to;
+    a.textContent = element.text;
+    a.href = element.a;
     const img = document.createElement('img');
-    img.src = listDataItem.img;
-    img.alt = listDataItem.alt;
+    img.src = element.img;
+    img.alt = element.alt;
     fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);  
   });
   ul.appendChild(fragment);
 }
-createList();
\ No newline at end of file
+
+const init = async () => {
+  createLoadingImage();
+  const response = await getListData();
+  const listData = await response.json();
+  createList(listData);
+  removeLoadingImage();
```

Comment:
順番逆にしたいです。loadingはデータ取得時から終わりまでを担当にしたいです
```suggestion
  removeLoadingImage();
    createList(listData);
```

Reply:
確かに、修正前のコードですとloadingは
「データ取得時からデータ取得し終わるまで」ではなく「データ取得時からリストを作り終えるまで」になってしまいますね・・・！修正しました！
(それぞれの関数がどの部分を担当するのかを考えながら実装する癖をつけたいと思います)

4eaf597
codesandbox(https://codesandbox.io/s/js-lesson11-dbwdt?file=/index.js)